lib/dataBroker.js

```diff
@@ -39,6 +40,28 @@ module.exports = function(streamDAO) {
           { statusCode: 400, message: util.format('Unknown type[%s], known types[%s]', datum.type, knownTypes) }
         );
       }
+      /*
+       * This is a hacky place to move this logic, but we need it to run
+       * before schema validation is attempted (or else it's pointless)
+       * so ¯\_(ツ)_/¯
+       */
+      if (datum.type === 'upload') {
+        var versionStr = datum.version.toLowerCase();
+
+        // TODO: longterm this check should be against a datamodel version
+        // and also probably not a hard go/no-go check
+        if (versionStr.indexOf('tidepool-uploader') !== -1 && uploadConfig.minimumUploaderVersion !== null ) {
```

Comment:
Yeah, good idea on the check for `uploadConfig.minimumUploaderVersion` - doing that and throwing a different error if there's no configured version (as that's also a situation we don't want to be in).

`tidepool-uploader` would not be in the version string if it's 3rd-party devs uploading data, or at least that was the original thought.